src/services/upload_repository.js

```diff
@@ -96,14 +96,14 @@ export default class UploadRepository {
     }
   }
 
-  async complementBundleMetadata(bundleMetadata) {
-    const bundleUploadData = await this.uploadsActions.getBundleUploadData(bundleMetadata.bundleId);
+  async composeBundleMetadataFromBlockchain(bundleId) {
```

Comment:
Just noted it's a bit strange it's in upload repository. Looks like it's used only on bundle download.

Reply:
It doesn't communicate with mongo, only takes bundle data from blockchain, so i guess it's ok, may be misleading though